tools/generator.js

```diff
@@ -0,0 +1,163 @@
+const Lang = imports.lang;
+const System = imports.system;
+
+const Gio = imports.gi.Gio;
+
+const Generator = new Lang.Class({
+    Name: 'Generator',
+
+    _init: function (dir_path) {
+        this._dir_path = dir_path;
+        this._sequence = 0;
+    },
+
+    _create_root: function () {
+        return {
+            'articles': [],
+            'images': [],
+            'sets': []
+        };
+    },
+
+    _create_article: function (path, title, mime_type, category, sequence) {
+        let tags = ['EknArticleObject'];
+
+        if (category)
+            tags.push(category);
+
+        return {
+            'contentType': mime_type,
+            'source': path,
+            'tags': tags,
+            'title': title,
+            'sequenceNumber': sequence,
+        };
+    },
+
+    _create_image: function (path, mime_type) {
+        let license = 'CC-BY 3.0';
```

Comment:
I think we could define some convention to specify this globally (e.g. having just 1 file with the license string for ALL of the contents) OR have per-file convention  (e.g. something in the name of the file or even having a separate file like article.pdf.license). Need to think more about that.

For now, I will just leave it empty (e.g. "").